addons/account/models/account_move.py

```diff
@@ -1762,14 +1762,22 @@ def _get_tax_cash_basis_base_key(self, tax, move, line):
         return (line.id, account_id.id, tax.id, line.currency_id.id, line.partner_id.id)
 
     def _get_tax_cash_basis_base_common_vals(self, key, new_move):
-        line, account_id, tax, currency_id, partner_id = key
-        line = self.env['account.move.line'].browse(line)
+        line_id, account_id, tax_id, currency_id, partner_id = key
+
+        line = self.env['account.move.line'].browse(line_id)
+        tax = self.env['account.tax'].browse(tax_id)
+
+        orig_inv_type = (self.debit_move_id + self.credit_move_id).mapped('invoice_id.type')
```

Comment:
are you sure this method is `ensure_one`?

Reply:
Yup, its only call is triggered by create_tax_cash_basis_entry (through an intermediate function called only once as well), which is ensure_one. I'll still add the check, though.